src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1272 @@
+                    self->invalidate();
+                    break;
+                }
+            }
+        }
+
+        // 0x00458721
+        static void on_scroll_mouse_over(ui::window* self, int16_t x, int16_t y, uint8_t scrollIndex)
+        {
+            auto index = getRowIndex(x, y);
+            uint16_t rowInfo = 0xFFFF;
+
+            for (auto i = 0; i < self->var_83C; i++)
+            {
+                rowInfo = self->row_info[i];
+                index--;
+                if (index < 0)
+                    break;
+            }
+            self->var_846 = rowInfo;
+            self->invalidate();
+            auto string = string_ids::buffer_337;
+
+            if (rowInfo == 0xFFFF)
+                string = string_ids::null;
+
+            if (stringmgr::get_string(string_ids::buffer_337)[0] != '\0')
+            {
+                if (string == self->widgets[common::widx::frame].tooltip)
+                {
+                    if (rowInfo == self->var_85C)
+                        return;
+                }
+            }
+            self->widgets[common::widx::frame].tooltip = string;
+            self->var_85C = rowInfo;
+            tooltip::closeAndReset();
+
+            if (rowInfo == 0xFFFF)
+                return;
+
+            auto industryObj = objectmgr::get<industry_object>(rowInfo);
+            const char* buffer = stringmgr::get_string(string);
+            char* ptr = (char*)buffer;
+            *ptr = '\0';
+
+            if (industryObj->producesCargo())
+            {
+                ptr = stringmgr::format_string(ptr, string_ids::industry_produces);
+                ptr = industryObj->getProducedCargoString(ptr);
+
+                if (industryObj->requiresCargo())
+                {
+                    ptr = stringmgr::format_string(ptr, string_ids::cargo_comma);
+                }
+            }
+
+            if (industryObj->requiresCargo())
+            {
+                ptr = stringmgr::format_string(ptr, string_ids::industry_requires);
+                ptr = industryObj->getRequiredCargoString(ptr);
+            }
+        }
+
+        // 0x004585B8
+        static void on_update(window* self)
+        {
+            if (!input::has_flag(input::input_flags::flag5))
+            {
+                auto cursor = input::getMouseLocation();
+                auto xPos = cursor.x;
+                auto yPos = cursor.y;
+                window* activeWindow = WindowManager::findAt(xPos, yPos);
+                if (activeWindow == self)
+                {
+                    xPos -= self->x;
+                    xPos += 26;
+                    yPos -= self->y;
+
+                    if ((yPos < 42) || (xPos <= self->width))
+                    {
+                        xPos = cursor.x;
+                        yPos = cursor.y;
+                        widget_index activeWidget = self->find_widget_at(xPos, yPos);
+
+                        if (activeWidget == widx::scrollview)
```

Comment:
In the original this was any widx after widx::panel this meant when you first clicked a tab it would increase the size. I think we should stick to that. Please change to `if(activeWidget > common::widx::panel)`